lib/AL_USDMaya/AL/usdmaya/nodes/ProxyDrawOverride.cpp

```diff
@@ -132,6 +138,7 @@ MUserData* ProxyDrawOverride::prepareForDraw(
 
   if(!data->m_shape || !data->m_shape->getRenderAttris(&data->m_params, frameContext, objPath))
   {
+    if (newData) delete data;
```

Comment:
This looks to be a little backward? 
Would it be better to see if fn.userNode() points to something valid, then create the data, if not, don't?

Reply:
If that doesn't remove the need for the newData bool, then I guess something along the lines of:

```c++
RenderUserData* newData = nullptr;
if(!data)
{
  data = newData = new RenderUserData;
}

// .... then later, just.... 
delete newData;

```